src/tools/build-manifest/src/main.rs

```diff
@@ -312,21 +326,17 @@ impl Builder {
             let filename = self.filename(pkgname, name);
             let digest = match self.digests.remove(&filename) {
                 Some(digest) => digest,
-                None => {
-                    return (name.to_string(), Target {
-                        available: false,
-                        url: None,
-                        hash: None,
-                        components: None,
-                        extensions: None,
-                    })
-                }
+                None => return (name.to_string(), Target::unavailable()),
             };
+            let xz_filename = filename.replace(".tar.gz", ".tar.xz");
```

Comment:
`xz_url: xz_digest.as_ref().map(|_| self.url(&xz_filename))` makes sure that the url is only specified if the digest is `Some`thing (and that its contents are based on the filename, not on the digest value).